src/globus_sdk/_testing/README.rst

```diff
@@ -0,0 +1,152 @@
+
+Once ``responses`` has been activated, each response fixture can be loaded and
+activated by name:
+
+.. code-block:: python
+
+    from globus_sdk._testing import load_response
+
+    # load_response will add the response to `responses` and return it
+    load_response("AuthClient.get_identities")
+    # "case" is used to have a single name map to multiple responses
+    data = load_response("AuthClient.get_identities", case="multiple")
+
+Responses can also be activated by passing an SDK client method, bound or
+unbound, as in:
+
+.. code-block:: python
+
+    import globus_sdk
+    from globus_sdk._testing import load_response
+
+    load_response(globus_sdk.AuthClient.get_identities)
+    load_response(globus_sdk.AuthClient.get_identities, case="unauthorized")
+
+    # or, with a bound method
+    ac = globus_sdk.AuthClient()
+    load_response(ac.AuthClient.get_identities, case="multiple")
+
+Activating "Scenarios"
+~~~~~~~~~~~~~~~~~~~~~~
+
+Some sets of fixtures may describe a scenario, and therefore it's desirable to
+load all of them at once:
+
+.. code-block:: python
+
+    from globus_sdk._testing import load_response_set
+
+    fixtures = load_response_set("scenario.foo")
+
+Getting Responses and ResponseSets without Activating
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to fetch a ``ResponseSet`` or ``RegisteredResponse`` without
+activating it, you can do this via the ``get_response_set`` method. Responses
+must always be part of a response set, and the default name for an individual
+response is ``"default"``.
+
+.. code-block:: python
+
+    from globus_sdk import AuthClient
+    from globus_sdk._testing import get_response_set
+
+    # rset will not be activated
+    rset = get_response_set(AuthClient.get_identities)
+    # you can get an individual response from rset
+    get_ids = rset.get("default")
+    # you can manually activate a whole set
+    rset.activate_all()
+    # or just one response from it by name
+    rset.activate("default")
+
+Note that activating a whole repsonse set may or may not make sense. For
+example, the response set for ``AuthClient.get_identities`` provides various
+responses for the same API call.
+
+Registering Response Sets
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can register your own response sets dynamically, and then load them up with
+the same ``load_response_set`` method. Note that custom response sets will
+override the builtin response sets, if names match.
+
+.. code-block:: python
+
+    from globus_sdk._testing import load_response_set, register_response_set
+    import uuid
+
+    # register a scenario under which Globus Auth get_identities and Globus
+    # Transfer operation_ls both return payloads of `{"foo": "bar"}`
+    # use an autogenerated endpoint ID and put it into the response metadata
+    # register_response_set takes dict data and converts it to fixtures
+    endpoint_id = str(uuid.uuid1())
+    register_response_set(
+        "foobar",
```

Comment:
I think it could be clearer in the doc here that while `"foobar"` is your own made-up tag for this, the `get_identities` etc. have to actually map to methods.

On that note, I sort of think it would be clearer organized like:
```python
{
  "auth": {
    "get_identities": ...,
  }
  "transfer": {
    etc.
  }
}
```

Reply:
Let me see what I can do about the doc to clarify in the next couple of days.

---

I also prefer the "service name" rather than the bare `__qualname__`.
How can we get from `AuthClient.get_identities` to `"auth", "get_identities"` best? Unbound methods are just functions (since py3), so `__qualname__` might still be best.

What about mapping from `__qualname__` to attributes of the `globus_sdk` top-level module, like so?
```python
name = func.__qualname__
module_name = f"globus_sdk._testing.data.{name}"
if "." in name:
    prefix, suffix = name.split(".")
    if hasattr(globus_sdk, prefix):
        service_name = getattr(globus_sdk, prefix).service_name
        module_name = f"globus_sdk._testing.data.{service_name}.{suffix}"
```

Reply:
I took this out into a dedicated function here: https://github.com/globus/globus-sdk-python/pull/517/commits/a1f083a440e2618fbd2dcf23845c4a2896674b19

As soon as CI passes, I plan to merge.